crates/services/consensus_module/poa/src/service.rs

```diff
@@ -582,16 +595,25 @@ where
     }
 }
 
-impl<T, B, I, S, PB, C> RunnableTask for MainTask<T, B, I, S, PB, C>
+impl<T, B, I, S, PB, C, TBP> RunnableTask for MainTask<T, B, I, S, PB, C, TBP>
 where
     T: TransactionPool,
     B: BlockProducer,
     I: BlockImporter,
     S: BlockSigner,
     PB: PredefinedBlocks,
     C: GetTime,
+    TBP: TriggerBlockProduction,
 {
     async fn run(&mut self, watcher: &mut StateWatcher) -> TaskNextAction {
+        tokio::select! {
+            biased;
+            _ = watcher.while_started() => {
+                return TaskNextAction::Stop
+            }
+            _ = self.block_production_trigger.wait_for_trigger() => {}
+        }
```

Comment:
This solution feels very brittle to me. I take it there's no good way to just wait for the other services to be started before starting this one without doing a major redesign of our service architecture?

Another evidence of why we need to rework our service traits and how we instantiate our services.

Reply:
i wouldn't call it major, we just need to add a `on_startup` trait method that's called before `start_and_await` for the services. we can block on this to ensure deterministic behaviour.

Reply:
Sure that could be an approach. It would also be nice if we could express the required preconditions for starting the PoA service explicitly, rather than relying on anyone instantiating the service implicitly knowing what's required before sending the start signal.